wordpress-mobile/WordPress-Android
# Plan Offers refactor companion.

This PR is a companion for [this](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2031)  FluxC PR and moves the PlanOffer related db tables to room. If all works well, there should be no changes in functionality.

### To test
All the following check should be done comparing side by side the UI behaviour from this PR with a previous release version of the app. Nothing should change visually.

- On a site that support plans (so no self-hosted nor P2s) go to My SIte
- Check you have the Plan item and tap on it
- Check you get the plans listed and they are correct
- PTR to check you get the same plans
- Tap on each plan and check the content is there and presented as expected
- Try to close the app or place it in background and coming back to the plan screens, the behaviours should always stay the same.
- Set airplane mode, come back to the plans and check you get the empty view, PTR still get the empty view
- Remove airplane mode and check the PTR is able to get the plans data and display them as expected.

## Regression Notes
1. Potential unintended areas of impact
Plan offers screens.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

3. What automated tests I added (or what prevented me from doing so)
Added unit tests to cover the room database functionality.


PR submission checklist:
